tests/hubspot_private_app/test_hubspot.py

```diff
@@ -0,0 +1,214 @@
+                    'archived_at': None,
+                    'associations': None,
+                    'created_at': datetime(2023, 2, 21, 9, 50, 48, 490000, tzinfo=timezone.utc),
+                    'id': '51',
+                    'properties': {
+                        'createdate': '2023-02-21T09:50:48.490Z',
+                        'email': 'bh@hubspot.com',
+                        'firstname': 'Brian',
+                        'hs_object_id': '51',
+                        'lastmodifieddate': '2023-02-21T09:50:55.765Z',
+                        'lastname': 'Halligan (Sample Contact)',
+                    },
+                    'properties_with_history': None,
+                    'updated_at': datetime(2023, 2, 21, 9, 50, 55, 765000, tzinfo=timezone.utc),
+                },
+            ],
+        }
+    )
+
+
+@pytest.fixture
+def hubspot_first_page(hubspot_full_page: _DictableDict) -> _DictableDict:
+    return _DictableDict(
+        **{
+            'paging': {
+                'next': {
+                    'after': '1',
+                    'link': 'https://api.hubapi.com/crm/v3/objects/contacts?limit=1&after=2',
+                }
+            },
+            'results': hubspot_full_page['results'][:1],
+        }
+    )
+
+
+@pytest.fixture
+def hubspot_second_page(hubspot_full_page: _DictableDict) -> _DictableDict:
+    return _DictableDict(paging=None, results=hubspot_full_page['results'][1:])
+
+
+@pytest.fixture
+def hubspot_all_results(hubspot_full_page: _DictableDict) -> list[_DictableDict]:
+    return [_DictableDict(r) for r in hubspot_full_page['results']]
+
+
+@pytest.fixture
+def hubspot_client(mocker: MockFixture) -> MagicMock:
+    basic_api = MagicMock()
+    api = MagicMock()
+    api.basic_api = basic_api
+
+    mocker.patch.object(HubspotConnector, '_api_for_dataset', return_value=api)
+
+    return api
+
+
+@pytest.fixture
+def expected_df() -> pd.DataFrame:
+    return pd.DataFrame(
+        {
+            'created_at': [
+                pd.Timestamp('2023-02-21 09:50:48.148000+0000', tz='UTC'),
+                pd.Timestamp('2023-02-21 09:50:48.490000+0000', tz='UTC'),
+            ],
+            'updated_at': [
+                pd.Timestamp('2023-02-21 09:50:53.651000+0000', tz='UTC'),
+                pd.Timestamp('2023-02-21 09:50:55.765000+0000', tz='UTC'),
+            ],
+            'id': ['1', '51'],
+            'createdate': ['2023-02-21T09:50:48.148Z', '2023-02-21T09:50:48.490Z'],
+            'email': ['emailmaria@hubspot.com', 'bh@hubspot.com'],
+            'firstname': ['Maria', 'Brian'],
+            'hs_object_id': ['1', '51'],
+            'lastmodifieddate': ['2023-02-21T09:50:53.651Z', '2023-02-21T09:50:55.765Z'],
+            'lastname': ['Johnson (Sample Contact)', 'Halligan (Sample Contact)'],
+        }
+    )
+
+
+@pytest.fixture
+def hubspot_connector() -> HubspotConnector:
+    return HubspotConnector(name='hubspot', access_token='s3cr3t')
```

Comment:
is it all it need to instantiate it ? 🤔 ... what about the "data_source_model" or we don't care... it's a fixture